gitpod-io/website
# 📊 Refactor: Better analytics for A/B Testing Experiments

## Description



This PR introduces a new Segment event (`component_loaded`). This Would be tied to certain components (explicitly). So that, we can better track the way that which component gets loaded. For this particular case (Git Signups CTA component). We've tied this that, is it getting rendered with which property `experiments_variant` which would have two possible values: `git_signup_cta_with_examples_loaded` OR `git_signup_cta_without_examples_loaded` depending on Feature Flag value (i.e. Examples Repo component is getting loaded or not).

It also updated the "Continue with GitHub" Clicks analytics, previously we were abusing `Position` ([code ref.](https://github.com/gitpod-io/website/blob/b2a9dfc0f3ad9b05f34a2f2da2912f15ea21f948/src/lib/components/index/get-started/link-git-repository.svelte#L10)). But, Now it would be tied to the same property `experiments_variant` (for consistency & easy to track) with values `git_signup_cta_with_examples_clicked` OR `git_signup_cta_without_examples_clicked` to eventually track the data in Graphs that **_How many times it got loaded vs How many times it got clicked_**.

Aside, This PR also Refactors the Feature Flag service and it's more reliable & Concrete with the feature flag value. 

## Related Issue(s)



Fixes MKWEB-28

## How to test



- Open Website Preview & Segment Staging Debugger.
- Open Homepage on Website. And you will able to see a new event named `component_loaded` in the segment with the property `git_signup_cta_experiment` and having value `git_signup_cta_with_examples_loaded` OR `git_signup_cta_without_examples_loaded`.
-  When you will click on "Continue with GitHub". It would trigger the `website_clicked` event with property `git_signup_cta_experiment` and its value as `git_signup_cta_with_examples_clicked` OR `git_signup_cta_without_examples_clicked`

## Release QA Checklist ✔️



<details>
  <summary>QA Checklist </summary>

## [Priority: High] Functional Testing



## [Priority: High] Content Testing


## [Priority: Medium] Performance Testing


## [Priority: Medium] Accessibility Testing


By including these key items in your website release QA checklist, you can ensure that your website is fully functional, user-friendly, and meets your business requirements.

## Security Testing

> **Note**:💡 If we are changing something related to cookies or data storing or analytics side.


  </details>


<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/3742"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
